packages/compass-collection/src/components/collection-header/collection-header.tsx

```diff
@@ -146,24 +82,50 @@ export const CollectionHeader: React.FunctionComponent<
   sourceName,
   editViewName,
   sourcePipeline,
-  stats,
 }) => {
   const darkMode = useDarkMode();
   const showInsights = usePreference('showInsights');
   const {
-    openCollectionsWorkspace,
     openCollectionWorkspace,
     openEditViewWorkspace,
+    openCollectionsWorkspace,
   } = useOpenWorkspace();
 
-  const ns = toNS(namespace);
-  const database = ns.database;
-  const collection = ns.collection;
+  const breadcrumbItems = useMemo(() => {
+    return [
+      // TODO (COMPASS-7684): add connection name
+      {
+        name: toNS(namespace).database,
+        onClick: () => openCollectionsWorkspace(toNS(namespace).database),
+      },
+      // When viewing a view, show the source namespace first
+      sourceName && {
+        name: toNS(sourceName).collection,
+        onClick: () => openCollectionWorkspace(sourceName),
+      },
+      // Show the current namespace
+      {
+        name: toNS(namespace).collection,
+        onClick: () => openCollectionWorkspace(namespace),
+      },
+      // When editing a view, show the view namespace last
```

Comment:
Not sure if it's something that needs addressing, but just want to call out that this logic doesn't match the logic in the workspace tab breadcrumb. Do we need to update it too?

![image](https://github.com/mongodb-js/compass/assets/5036933/bce6bf20-0fcc-4588-8545-579b5af2c38d)

Reply:
That's a really good point. I'll check with design and if needed create a follow up.